E2E Test Case `TestDeletePod` Failed on CentOS 7

[Matrix compatibility test job](https://jenkins.antrea-ci.rocks/job/antrea-weekly-matrix-compatibility-test/) runs weekly, and the E2E test running under CentOS 7 fails on `TestDeletePod` test case in recent two weeks:

```
=== RUN   TestDeletePod
--- FAIL: TestDeletePod (35.60s)
    fixtures.go:107: Creating 'antrea-test' K8s Namespace
    fixtures.go:70: Applying Antrea YAML
    fixtures.go:74: Waiting for all Antrea DaemonSet Pods
    fixtures.go:78: Checking CoreDNS deployment
    basic_test.go:169: Creating a busybox test Pod on 'matrix-centos-7-k8s-v1-17-5-build-num-3-tj6v4'
    basic_test.go:83: The Antrea Pod for Node 'matrix-centos-7-k8s-v1-17-5-build-num-3-tj6v4' is 'antrea-agent-wmrm6'
    basic_test.go:99: Host interface name for Pod is 'test-pod-9935a2'
    basic_test.go:129: Checking that the veth interface and the OVS port exist
    basic_test.go:131: Interface 'test-pod-9935a2' does not exist on Node 'matrix-centos-7-k8s-v1-17-5-build-num-3-tj6v4'
    basic_test.go:140: Deleting Pod 'test-pod-hs08t57g'
    basic_test.go:145: Checking that the veth interface and the OVS port no longer exist
    fixtures.go:167: Exporting test logs to '/var/lib/jenkins/workspace/antrea-weekly-matrix-compatibility-test/IS_MATRIX_TEST/True/K8S_VERSION/v1.17.5/TEST_OS/centos-7/labels/antrea-test-node/antrea-test-logs/TestDeletePod/beforeTeardown.Nov01-03-00-57'
    fixtures.go:265: Error when exporting kubelet logs: error when running journalctl on Node 'matrix-centos-7-k8s-v1-17-5-build-num-3-tj6v4', is it available?
fixtures.go:274: Deleting 'antrea-test' K8s Namespace
```

--- a/test/e2e/basic_test.go
+++ b/test/e2e/basic_test.go
@@ -110,13 +110,15 @@ func (data *TestData) testDeletePod(t *testing.T, podName string, nodeName strin
 	t.Logf("Host interface name for Pod is '%s'", ifName)
 
 	doesInterfaceExist := func() bool {
-		cmd := fmt.Sprintf("ip link show %s", ifName)
-		if rc, _, _, err := RunCommandOnNode(nodeName, cmd); err != nil {
-			t.Fatalf("Error when running ip command on Node '%s': %v", nodeName, err)
-		} else {
-			return rc == 0
+		cmd := []string{"ip", "link", "show", ifName}
+		stdout, stderr, err := data.runCommandFromPod(antreaNamespace, antreaPodName, agentContainerName, cmd)
+		if err != nil {
+			if strings.Contains(stderr, "does not exist") {
+				return false
+			}
+			t.Fatalf("Error when running ip command in Pod '%s': %v - stdout: %s - stderr: %s", antreaPodName, err, stdout, stderr)
 		}
-		return false
+		return true
 	}
 
 	doesOVSPortExist := func() bool {
@@ -331,18 +333,16 @@ func testReconcileGatewayRoutesOnStartup(t *testing.T, data *TestData, isIPv6 bo
 	}
 
 	getGatewayRoutes := func() (routes []Route, err error) {
-		var cmd string
+		var cmd []string
 		if !isIPv6 {
-			cmd = fmt.Sprintf("ip route list dev %s", antreaGWName)
+			cmd = []string{"ip", "route", "list", "dev", antreaGWName}
 		} else {
-			cmd = fmt.Sprintf("ip -6 route list dev %s", antreaGWName)
+			cmd = []string{"ip", "-6", "route", "list", "dev", antreaGWName}
 		}
-		rc, stdout, _, err := RunCommandOnNode(nodeName, cmd)
+		podName := antreaPodName()
+		stdout, stderr, err := data.runCommandFromPod(antreaNamespace, podName, agentContainerName, cmd)
 		if err != nil {
-			return nil, fmt.Errorf("error when running ip command on Node '%s': %v", nodeName, err)
-		}
-		if rc != 0 {
-			return nil, fmt.Errorf("running ip command on Node '%s' returned error", nodeName)
+			return nil, fmt.Errorf("error when running ip command in Pod '%s': %v - stdout: %s - stderr: %s", podName, err, stdout, stderr)
 		}
 		re := regexp.MustCompile(`([^\s]+) via ([^\s]+)`)
 		for _, line := range strings.Split(stdout, "\n") {

